### ISSUE TITLE
Prediction of Inputs that achieve outputs conditions

### ISSUE THREAD

**Issue by alks04** [keras-team/keras#10527]
Hi everybody, im learning about how to make predictions using keras and tensorflow, the next code is a training code to learn about AI.

`import` numpy as np 
`from` keras.models import Sequential
`from` keras.layers.core import Dense

`#Inputs` to be learned
`training_data=np.array([[0,0,0],[0,0,1],[0,1,0],[0,1,1],[1,0,0],[1,0,1],[1,1,0],[1,1,1]],"float32")`

`#Prediction` array to be used once the model is generated.
`test1=np.array([[0,1,0],[1,1,1],[0,0,0],[1,0,0]],"float32")`

`#Results` (outputs) of each input to be learned
`target_data=np.array([[0],[1],[1],[0],[1],[0],[0],[0]],"float32")`

`model=Sequential()`
`model.add(Dense(50,input_dim=3,activation='relu'))`
`model.add(Dense(1,activation='sigmoid'))`

`model.compile(loss='mean_squared_error',optimizer='adam',metrics=['binary_accuracy'])`
`model.fit(training_data,` target_data, epochs=1000)

`#Evaluation` of model
`scores=model.evaluate(training_data,target_data)`
`#Result` of the model
`print("\n%s:` %.2f%%" % (model.metrics_names[1],scores[1]*100))
`#Test` of the model using test1 to predict.
`print(model.predict(test1).round())`

What I'm trying to learn is about how to print the inputs that achieve the desired outputs condition. 

For example, when the desired output is "1": 
`print(model.predict(target_data(output=1)))`

the printed result shall be
[0,0,1],[0,1,0],[1,0,1]

i already read the keras manual and more information in differents websites but anyone of them works when trying to predict the inputs results using an output value, im not sure if this is possible with keras or with other library.
Can you help me in this one? 
Thank you!
